File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = float('inf')
        max_profit = 0 
        for i in range(len(prices)):
            min_price = min(prices[i], min_price)
            current_profit = prices[i] - min_price
            max_profit = max(max_profit, current_profit)

        return max_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int max = 0;
        for (int p:prices){
            if (p < min){
                min = p; 
            } else {
                int profit = p - min;
                max = Math.max(profit, max);
            }
        }
        return max;
    }
}